framework/src/media/audio/audio_manager.c

```diff
@@ -906,13 +906,21 @@ audio_manager_result_t stop_audio_stream_out(void)
 
 	pthread_mutex_lock(&(g_audio_out_cards[g_actual_audio_out_card_id].card_mutex));
 
-	ret = pcm_drain(g_audio_out_cards[g_actual_audio_out_card_id].pcm);
+	if (g_audio_out_cards[g_actual_audio_out_card_id].status == AUDIO_CARD_PAUSE) {
+		ret = pcm_drop(g_audio_out_cards[g_actual_audio_out_card_id].pcm);
+	} else {
+		ret = pcm_drain(g_audio_out_cards[g_actual_audio_out_card_id].pcm);
+	}
```

Comment:
below log shows only pcm_drain, checks only, and I think pcm_close should be called even if pcm_drain failed.
bool isPaused = g_audio_out_cards[g_actual_audio_out_card_id].status == AUDIO_CARD_PAUSE;
if (isPaused) {
   ret = pcm_drop();
   if (ret < 0) {
        meddbg("pcm_drop failed, ret : %d\n", ret);
   }
} else {
   ret = pcm_drain();
   if (ret < 0) {
       meddbg("pcm_drain failed ret : %d\n", ret);
    }
    ret = pcm_close() //try close even if it failed to next stream out.
   if (ret < 0) {
        meddbg("pcm_close failed, ret : %d\n", ret);
   }
}

pthread_mutex_unlock();
if (ret < 0) {
    return AUDIO_MANAGER_DEVICE_FAIL;
}
card.....
```

Reply:
Okay, modified!